src/geocat/f2py/rcm2points_wrapper.py

```diff
@@ -0,0 +1,207 @@
+	    of fi. The latitude order must be west-to-east.
+	fi (:class:`numpy.ndarray`):
+	    A multi-dimensional array to be interpolated. The rightmost two
+	    dimensions (latitude, longitude) are the dimensions to be interpolated.
+	lat1dPoints (:class:`numpy.ndarray`):
+	    A one-dimensional array that specifies the latitude coordinates of
+	    the output locations.
+	lon1dPoints (:class:`numpy.ndarray`):
+	    A one-dimensional array that specifies the longitude coordinates of
+	    the output locations.
+    msg (:obj:`numpy.number`):
+	    A numpy scalar value that represent a missing value in fi.
+	    This argument allows a user to use a missing value scheme
+	    other than NaN or masked arrays, similar to what NCL allows.
+	opt (:obj:`numpy.number`):
+	    opt=0 or 1 means use an inverse distance weight interpolation.
+	    opt=2 means use a bilinear interpolation.
+        
+    Returns:
+        
+    	:class:`numpy.ndarray`: The interpolated grid. A multi-dimensional array
+    	of the same size as fi except that the rightmost dimension sizes have been
+    	replaced by the number of coordinate pairs (lat1dPoints, lon1dPoints).
+    	Double if fi is double, otherwise float.
+    
+    Description:
+        
+    	Interpolates data on a curvilinear grid, such as those used by the RCM (Regional Climate Model),
+    	WRF (Weather Research and Forecasting) and NARR (North American Regional Reanalysis)
+    	models/datasets to an unstructured grid. All of these have latitudes that are oriented south-to-north.
+    	A inverse distance squared algorithm is used to perform the interpolation.
+    	Missing values are allowed and no extrapolation is performed.
+    '''
+    # Basic sanity checks
+    if yi.shape[0] != xi.shape[0] or yi.shape[1] != xi.shape[1]:
+        raise DimensionError(
+            "ERROR rcm2points: The input lat/lon grids must be the same size !")
+
+    if yo.shape[0] != xo.shape[0]:
+        raise DimensionError(
+            "ERROR rcm2points: The output lat/lon grids must be same size !")
+
+    if yi.shape[0] < 2 or xi.shape[0] < 2 or yi.shape[
+            1] < 2 or xi.shape[1] < 2:
+        raise DimensionError(
+            "ERROR rcm2points: The input/output lat/lon grids must have at least 2 elements !"
+        )
+
+    if fi.ndim < 2:
+        raise DimensionError(
+            "ERROR rcm2points: fi must be at least two dimensions !\n")
+
+    if fi.shape[fi.ndim - 2] != yi.shape[0] or fi.shape[fi.ndim -
+                                                           1] != xi.shape[1]:
+        raise DimensionError(
+            "ERROR rcm2points: The rightmost dimensions of fi must be (nlat2d x nlon2d),"
+            "where nlat2d and nlon2d are the size of the lat2d/lon2d arrays !")
+    
+    if (xi is None) | (yi is None):
+         raise CoordinateError(
+             "rcm2points: xi and yi should always be provided")
+   
+
+    # ''' Start of boilerplate
```

Comment:
At this point, I'd like to know @pilotchute 's thoughts. "boilerplate" is supposed to prepare input-output data for Dask (chunking, etc.), right? We also need to make some sanity checks as above to see if the inputs are matching requirements. Are you fine with having them directly in this outer wrapper (like done in this PR), or should we address such sanity checks in a separate utility function  (and just call it here) in order not to hammer the readibility?

Reply:
sanity checks should be pulled out into a module, much like we did with `missing_values`.